dataplicity/m2m/fileservice.py

```diff
@@ -0,0 +1,77 @@
+"""
+
+An M2M Service to retrieve files.
+
+File data is sent in chunks.
+
+"""
+
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from functools import partial
+import logging
+import threading
+
+from lomond.errors import WebSocketError
+
+
+log = logging.getLogger('m2m')
+
+
+class FileService(threading.Thread):
+    """A thread the sends data from a file over m2m."""
+
+    # A word on lifetime management of this and similar objects...
+    # There does not need to be any other references to these objects;
+    # the thread maintains a reference to a m2m channel in the run
+    # function, so the entire object will be garbage collected when
+    # the thread exits.
+
+    CHUNK_SIZE = 4096
+
+    def __init__(self, channel, path):
+        self._repr = "FileService({!r}, {!r})".format(channel, path)
+        super(FileService, self).__init__(
+            args=(channel, path),
+            target=self.run_service
+        )
+        self.start()
+
+    def __repr__(self):
+        return self._repr
+
+    def run_service(self, channel, path):
+        """Run the thread and log exceptions."""
+        try:
+            self._run_service(channel, path)
+        except Exception:
+            log.exception("error running %r", self)
+
+    def _run_service(self, channel, path):
```

Comment:
Right... on the second look, I think it's my fault entirely. Initially I thought that the CommandService was the base class for the rest of the commands and so when I saw two functions with identical names, I assumed that they were overrides. I can now see that these are completely separate services. Sorry for confusion :(